File: src/FamilyOs.Application/Auth/Dtos/CurrentUserDto.cs
namespace FamilyOs.Application.Auth.Dtos;

public record CurrentUserDto(
    Guid UserAccountId,
    Guid? FamilyMemberId,
    string DisplayName,
    string Email,
    string Role,
    UserPreferencesDto? Preferences);

public record UserPreferencesDto(
    bool EmailEnabled,
    string? QuietHoursStart,
    string? QuietHoursEnd);

File: frontend/src/app/core/auth/current-user.dto.ts
export interface UserPreferencesDto {
  emailEnabled: boolean;
  quietHoursStart: string | null;
  quietHoursEnd: string | null;
}

export interface CurrentUserDto {
  userAccountId: string;
  familyMemberId: string | null;
  displayName: string;
  email: string;
  role: 'Admin' | 'Adult' | 'Child';
  preferences: UserPreferencesDto | null;
}
